src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+            valueCopy = new TextNode("true")
+          } else {
+            valueCopy = new TextNode("false")
+          }
+        }
+        case _ => {}
+      }
+      if (Array[String]("true", "false", "start", "end").contains(valueCopy.asText())) {
+        (valueCopy, Array[String](), PropertyType.Dialect)
+      } else {
+        (BooleanNode.getTrue, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Dialect)
+      }
+    }
+  }
+
+  def columnsProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def columnReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode => {
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s)
+          (arrayNode, Array[String](), csvwPropertyType)
+        }
+        case a: ArrayNode => (a, Array[String](), csvwPropertyType)
+      }
+    }
+  }
+
+  def targetFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def scriptFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def sourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def resourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def schemaReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      val url = baseUrl + value.asText()
+      // Don't know how to place a URI object in JsonNode, keeping the text value as of now
+      (new TextNode(url), Array[String](), csvwPropertyType)
+    }
+  }
+
+  def dialectProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case o: ObjectNode => {
+          val valueCopy = o.deepCopy()
+          var warnings = Array[String]()
+          val fieldsAndValues = Array.from(valueCopy.fields.asScala)
+          for (fieldsAndValue <- fieldsAndValues) {
```

Comment:
I'd say `fieldAndValue` is the singular.

Reply:
Done